regression/Makefile

```diff
@@ -20,15 +20,37 @@ DIRS = ansi-c \
        test-script \
        # Empty last line
 
-# Check for the existence of $dir. Tests under goto-gcc cannot be run on
-# Windows, so appveyor.yml unlinks the entire directory under Windows.
+# Run all test directories in sequence
+.PHONY: test
 test:
 	@for dir in $(DIRS); do \
-	  if [ -d "$$dir" ]; then \
-	    $(MAKE) -C "$$dir" test || exit 1; \
-	  fi; \
+        $(MAKE) "test-$$dir" || exit 1; \
 	done;
 
+# Pattern to execute a single directory, checking for existence of the directory
+# first. Tests under goto-gcc cannot be run on Windows, so appveyor.yml unlinks
+# the entire directory under Windows.
+test-%:
+	if [ -d "$*" ]; then \
+		echo "Running $*..." ; \
+		$(MAKE) -C "$*" test || exit 1; \
+	fi;
```

Comment:
See above for how to avoid the extra test here, at which point this rule could be simplified?